pkg/prometheus/promcfg_test.go

```diff
@@ -9421,6 +9421,54 @@ scrape_configs:
   metric_relabel_configs:
   - regex: noisy_labels.*
     action: labeldrop
+`,
+		},
+		{
+			name: "dns_sd_config-srv-record",
+			scSpec: monitoringv1alpha1.ScrapeConfigSpec{
+				DNSSDConfigs: []monitoringv1alpha1.DNSSDConfig{
+					{
+						Names: []string{"web.example.com"},
+					},
+				},
+			},
+			expectedCfg: `global:
```

Comment:
Do you want to move these to golden files now or in a follow up? (see https://github.com/prometheus-operator/prometheus-operator/pull/5873)

Reply:
I think it worth!
If you want I can add to the open PR all the changes to this test and you just need to change you specific new test case.

Reply:
I will add to this PR so I can familiarize with it too :)

Reply:
Since tests are already passed and if no other change is required, I can do it in follow up PR so save some ci cycles. If any other change needed I can add it in this PR itself. wdyt?

Reply:
I had 2 minor comments on this PR so I merged #5873 and @slashpai can use golden files for the next iteration.

Reply:
I would need to update for all tests in `TestScrapeConfigSpecConfig`, should we add those changes in this? I can have a follow up PR after addressing the comments to keep all changes in feature?

Reply:
I'm fine having it as a follow-up.